hazelcast/src/main/java/com/hazelcast/management/TimedMemberStateFactory.java

```diff
@@ -175,46 +174,53 @@ private void createMemState(MemberStateImpl memberState,
         }
     }
 
-    private int handleExecutorService(MemberStateImpl memberState, int count, Config config, IExecutorService executorService) {
-        if (config.findExecutorConfig(executorService.getName()).isStatisticsEnabled()) {
-            LocalExecutorStatsImpl stats = (LocalExecutorStatsImpl) executorService.getLocalExecutorStats();
-            memberState.putLocalExecutorStats(executorService.getName(), stats);
-            return count + 1;
+    private int handleExecutorService(MemberStateImpl memberState, int count, Config config,
+                                      Map<String, LocalExecutorStats> executorServices) {
+        for (String name : executorServices.keySet()) {
```

Comment:
- it would be better to get entryset and then iterate over it
- if cout >= maxVisibleInstanceCount you can stop iterating and exit the method, no need to continue iterating